lib/evse_security/crypto/openssl/openssl_supplier.cpp

```diff
@@ -591,7 +590,24 @@ CertificateValidationResult OpenSSLSupplier::x509_verify_certificate_chain(X509H
         }
     }
 
-    X509_STORE_CTX_init(store_ctx_ptr.get(), store_ptr.get(), get(target), NULL);
+    X509_STACK_UNSAFE_ptr untrusted = nullptr;
+
+    // Build potentially untrusted intermediary (subca) certificates
+    if (false == untrusted_subcas.empty()) {
+        untrusted = X509_STACK_UNSAFE_ptr(sk_X509_new_null());
+
+        for (auto& untrusted_cert : untrusted_subcas) {
+            if (false == sk_X509_push(untrusted.get(), get(untrusted_cert))) {
```

Comment:
X509_add_cert() could be used instead. supports useful flags:
X509_ADD_FLAG_NO_DUP and X509_ADD_FLAG_NO_SS
Prevents duplicates and self signed certificates from being added

Reply:
it only needs to be done once. My preference is to do the check in the OpenSSL code.
It depends whether OpenSSLSupplier::x509_verify_certificate_chain() is called directly or not.